crates/sui-core/src/authority_server.rs

```diff
@@ -1225,6 +1225,19 @@ impl ValidatorService {
                                 return None;
                             }
                             let contents_len = header_contents.len();
+                            // Network topology should not be very dynamic, therefore if it changes and the above
+                            // invariant is violated, we should fail loudly so that the node config can be updated.
+                            assert!(
```

Comment:
I agree with the goal of trying to get people to fix their configs but I think an assert is to heavy handed here. Can we use an ERROR log, and possibly a flag metric that we can monitor?